go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -262,16 +262,15 @@ func (c *Controller) convertANPPeersToIPs(anp *adminNetworkPolicyState) error {
 	return nil
 }
 
-// convertANPPeersToIPSet creates a set of peerIPs for all the peers passed as argument
+// expandANPRulePeersToIPSet creates a set of peerIPs for all the peers passed as argument
 // This function also takes care of populating the adminNetworkPolicyPeer.namespaces cache
-// It also adds up all the peerIPs that are supposed to be present in the created AS and returns them on
-// a per-rule basis so that the actual ops to transact these into the AS can be constructed using that
-func (c *Controller) convertANPPeersToIPSet(peers []*adminNetworkPolicyPeer) (sets.Set[string], error) {
-	peerIPs := sets.Set[string]{}
+// It also adds up all the peerIPs that are supposed to be present in the created AddressSet and returns them on
+// a per-rule basis so that the actual ops to transact these into the AddressSet can be constructed using that
+func (c *Controller) expandANPRulePeersToIPSet(peers []*adminNetworkPolicyPeer, peerIPs sets.Set[string]) error {
```

Comment:
ack that works for me, I am terrible at namings so appreciate the perspective here. I was about to do that in commit3 changing this to `expandANPRulePeers` because that's when CIDRs take entry and until now even if its nodes, namespaces, pods they are all expanded and still resolved to IPs, so will do this as part of commit3.

Reply:
this is done in commit3!

Reply:
Considering the name of the caller as well (forgot you asked about this as well), I would probably have then

caller
```
func (c *Controller) expandANPRulePeers(anp *adminNetworkPolicyState) error
```

this func
```
func (c *Controller) expandRulePeers(gressRule *gressRule) error
```

Reply:
i think i got right finally in the latest push, if not let me know..